### PR TITLE
[Merge to feature/GEODE-7665] GEODE-9195: Remove PR clear local locking

### PR DESCRIPTION
**Pull request by kirklund**
1. Remove PR clear local locking
2. Unit test changes in BucketRegion and DistributedRegion
3. Unit test most of PartitionedRegionClearMessage

### CODE REVIEW

**Review by nabarunnag — APPROVED** [review]

File: geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionClearMessageTest.java
Comment (DonalEvans) [inline geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionClearMessageTest.java:108]: This test case seems to be covered in the test above it. Is it necessary to have a seperate test for it here too?

File: geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionClearMessageTest.java
Comment (DonalEvans) [inline geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionClearMessageTest.java:121]: This test case seem to be covered in `construction_findsAllDependencies()`. Is it necessary to test it here too?

**Review by DonalEvans — COMMENTED** [review]
A few questions about the intended impact of these changes, and some potentially duplicate test cases.

File: geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java
Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java:608]: I'm trying to clean up code on feature/GEODE-7665 and write unit tests where we have none yet. The team wants to review code in PR form before merging to the feature branch which unfortunately pulls in CODEOWNERS and requires CO approvals. This feature branch is still wip and is not going to develop.

**Comment by kirklund** [comment]
Closing PR because I don't want to waste the time of CODEOWNERs to handle a PR that is going to feature/GEODE-7665 instead of develop. I also have lots of conflicts to handle again.

**Comment by onichols-pivotal** [comment]
Actually, your feature branch does not *require* codeowners.  You can merge PRs on any branch other than develop with 0 reviews if you want.

However, since your last rebase with develop picked up the CODEOWNERS file, GitHub does show who would be the relevant codeowners for this PR if it were actually for develop.

**Pull request closed without merging**
